orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -213,13 +236,24 @@ Status AddOrSkipRecordForwardWaitBackward(Graph& graph, Node* send_fw, Node* rec
                   {},          /* attribute */
                   kMSDomain);
     wait_node = &new_node;
-    ORT_UNUSED_PARAMETER(wait_node);
+
+    backward_waited_event_name = wait_node->InputDefs()[0]->Name();
+    backward_output_name = wait_node->OutputDefs()[0]->Name();
   }
 
   return Status::OK();
 }
 
-Status TransformGraphForPipeline(Graph& graph) {
+Status TransformGraphForPipeline(
+  Graph& graph,
+  std::string& forward_waited_event_name,
+  std::string& forward_recorded_event_name,
+  std::string& backward_waited_event_name,
+  std::string& backward_recorded_event_name,
+  std::string& forward_waited_output_name,
+  std::string& forward_recorded_output_name,
+  std::string& backward_waited_output_name,
+  std::string& backward_recorded_output_name) {
```

Comment:
These names looks a bit mouthful and not consistant with the rest naming style. How about:

  std::string& wait_fw_event_name,
  std::string& record_fw_event_name,
  std::string& wait_bw_event_name,
  std::string& record_bw_event_name,
  std::string& wait_fw_output_name,
  std::string& record_fw_output_name,
  std::string& wait_bw_output_name,
  std::string& record_bw_output_name

Reply:
If I must choose a shorter name, I would do `fw_wait_event_name`. The name of larger scope should come earlier like `onnx.operator.wait` instead of onnx.wait.operator`.

Reply:
Because we are adding more events, I'd like to keep names as detailed as possible and simplify them once we have all.